ext/mysqli/tests/mysqli_begin_transaction.phpt

```diff
@@ -85,14 +85,30 @@ if (!have_innodb($link))
 
     if (mysqli_get_server_version($link) >= 50605) {
         /* does it like stupid names? */
-        if (@!$link->begin_transaction(MYSQLI_TRANS_START_READ_WRITE, "*/trick me?\n\0"))
-                printf("[020] [%d] %s\n", mysqli_errno($link), mysqli_error($link));
+        if (@!$link->begin_transaction(MYSQLI_TRANS_START_READ_WRITE, "*/trick me?\n\0")) {
+            printf("[020] [%d] %s\n", mysqli_errno($link), mysqli_error($link));
+        } else {
+            mysqli_rollback($link);
+        }
 
         /* does it like stupid names? */
-        if (@!$link->begin_transaction(MYSQLI_TRANS_START_READ_WRITE, "az09"))
+        if (@!$link->begin_transaction(MYSQLI_TRANS_START_READ_WRITE, "az09")) {
             printf("[021] [%d] %s\n", mysqli_errno($link), mysqli_error($link));
+        } else {
+            mysqli_rollback($link);
+        }
     }
 
+    mysqli_report(MYSQLI_REPORT_ERROR|MYSQLI_REPORT_STRICT);
+    try {
+        mysqli_begin_transaction($link);
+        mysqli_begin_transaction($link);
+    } catch (\mysqli_sql_exception $e) {
+        echo $e->getMessage() . \PHP_EOL;
+    }
+
+    mysqli_rollback($link);
```

Comment:
```suggestion
    mysqli_commit($link);
```

to test if the first `mysqli_begin_transaction` can still be commit

Reply:
The result is the same no matter which one you use. This is because the only errors that occur are due to the checks added this time.

Personally, I'm fine with either.